crates/rpc/rpc-eth-types/src/error/mod.rs

```diff
@@ -23,9 +23,22 @@ use revm::context_interface::result::{
     EVMError, ExecutionResult, HaltReason, InvalidHeader, InvalidTransaction, OutOfGasError,
 };
 use revm_inspectors::tracing::MuxError;
-use std::convert::Infallible;
+use std::{any::Any, convert::Infallible};
 use tracing::error;
 
+/// Extension trait for `InvalidTxError` to provide casting functionality
+pub trait InvalidTxErrorExt {
+    /// Try to cast the dyn `InvalidTxError` to `InvalidTransaction`
+    fn as_invalid_tx_err(&self) -> Option<&InvalidTransaction>;
+}
+
+impl InvalidTxErrorExt for dyn InvalidTxError {
+    fn as_invalid_tx_err(&self) -> Option<&InvalidTransaction> {
+        let any_ref: &dyn Any = self;
+        any_ref.downcast_ref::<InvalidTransaction>()
+    }
+}
```

Comment:
this doesnt really belong here and we should rather add this to the existing InvalidTxError trait in alloy-evm

because the way this behaves rn is a bit pointless since its impled on the dyn